Material.Blazor/Foundation/ComponentFoundation.cs

```diff
@@ -23,7 +23,9 @@ public abstract class ComponentFoundation : ComponentBase, IDisposable
         private static readonly ImmutableArray<string> EssentialSplattableAttributes = ImmutableArray.Create("formnovalidate", "max", "min", "role", "step", "tabindex", "type", "data-prev-page");
         private bool? disabled = null;
 
-        [Inject] private protected IBatchingJSRuntime JsRuntime { get; set; }
+        [Inject] private IBatchingJSRuntime InjectedJsRuntime { get; set; }
+        protected IBatchingJSRuntime JsRuntime { get; set; }
+        [CascadingParameter] private MBDialog ParentDialog { get; set; }
```

Comment:
If we're inside a dialog, we'll know with this cascading parameter. We "override" the JsRuntime in that case